x-pack/plugins/security_solution/public/timelines/components/timeline/query_bar/eql/index.tsx

```diff
@@ -0,0 +1,200 @@
+    const getOptionsSelected = useMemo(() => getEqlOptions(), []);
+    const optionsSelected = useDeepEqualSelector((state) => getOptionsSelected(state, timelineId));
+
+    const { loading: indexPatternsLoading, indexPattern, selectedPatterns } = useSourcererScope(
+      SourcererScopeName.timeline
+    );
+
+    const initialState = {
+      ...defaultValues,
+      index: selectedPatterns.sort(),
+      eqlQueryBar: {
+        ...defaultValues.eqlQueryBar,
+        query: { query: expression, language: 'eql' },
+      },
+    };
+
+    const { form } = useForm<TimelineEqlQueryBar>({
+      defaultValue: initialState,
+      options: { stripEmptyFields: false },
+      schema,
+    });
+    const { getFields } = form;
+
+    const applyLanguageFilterQuery = useCallback(
+      (language: string) =>
+        dispatch(
+          timelineActions.applyKqlFilterQuery({
+            id: timelineId,
+            filterQuery: {
+              kuery: {
+                kind: language as KueryFilterQueryKind,
+                expression: '',
+              },
+              serializedQuery: '',
+            },
+          })
+        ),
+      [dispatch, timelineId]
+    );
+
+    const onOptionsChange = useCallback(
+      (field: FieldsEqlOptions, value: string | null) =>
+        dispatch(
+          timelineActions.updateEqlOptions({
+            id: timelineId,
+            field,
+            value,
+          })
+        ),
+      [dispatch, timelineId]
+    );
+
+    const [{ eqlQueryBar: formEqlQueryBar }] = useFormData<TimelineEqlQueryBar>({
+      form,
+      watch: ['eqlQueryBar'],
+    });
+
+    const optionsData = useMemo(
+      () => ({
+        keywordFields: indexPattern.fields
+          .filter((f) => f.esTypes?.includes('keyword'))
+          .map((f) => ({ label: f.name })),
+        dateFields: indexPattern.fields
+          .filter((f) => f.type === 'date')
+          .map((f) => ({ label: f.name })),
+        allFields: indexPattern.fields
+          .filter((f) => f.type !== 'date')
+          .map((f) => ({ label: f.name })),
+      }),
+      [indexPattern]
+    );
+
+    useEffect(() => {
+      const { index: indexField } = getFields();
+      const newIndexValue = selectedPatterns.sort();
+      const indexFieldValue = (indexField.value as string[]).sort();
+      if (!isEqual(indexFieldValue, newIndexValue)) {
+        indexField.setValue(newIndexValue);
+      }
+    }, [getFields, selectedPatterns]);
+
+    useEffect(() => {
+      if (
+        formEqlQueryBar != null &&
+        !isEmpty(formEqlQueryBar.query.query.trim()) &&
+        isQueryBarValid &&
+        !isQueryBarValidating
+      ) {
+        dispatch(
+          timelineActions.applyKqlFilterQuery({
+            id: timelineId,
+            filterQuery: {
+              kuery: {
+                kind: formEqlQueryBar.query.language as KueryFilterQueryKind,
+                expression: formEqlQueryBar.query.query as string,
```

Comment:
consider replacing this `as` cast the with the following if it makes sense here:

```
expression: `${formEqlQueryBar.query.query}`,
```